client/blocks/login/index.jsx

```diff
@@ -240,57 +241,61 @@ class Login extends Component {
-							{
-								components: {
-									a: (
-										<a
-											href="https://woocommerce.com/2017/01/woocommerce-requires-wordpress-account/"
-											target="_blank"
-											rel="noopener noreferrer"
-										/>
-									),
-									br: <br />,
-								},
-							}
-						) }
-					</p>
-				);
-			}
-
-			if ( isWooOAuth2Client( oauth2Client ) && wccomFrom ) {
-				preHeader = (
-					<Fragment>
-						{ 'cart' === wccomFrom ? (
-							<WooCommerceConnectCartHeader />
-						) : (
-							<div className="login__woocommerce-wrapper">
-								<div className={ classNames( 'login__woocommerce-logo' ) }>
-									<svg width={ 200 } viewBox={ '0 0 1270 170' }>
-										<AsyncLoad
-											require="calypso/components/jetpack-header/woocommerce"
-											darkColorScheme={ false }
-											placeholder={ null }
-										/>
-									</svg>
+			if ( isWooOAuth2Client( oauth2Client ) ) {
+				if ( isPartnerSignup ) {
+					preHeader = <Gridicon icon="my-sites" size={ 72 } />;
+					headerText = translate( 'Log in to your account' );
+					postHeader = null;
+				} else if ( wccomFrom ) {
+					preHeader = (
+						<Fragment>
+							{ 'cart' === wccomFrom ? (
+								<WooCommerceConnectCartHeader />
+							) : (
+								<div className="login__woocommerce-wrapper">
+									<div className={ classNames( 'login__woocommerce-logo' ) }>
+										<svg width={ 200 } viewBox={ '0 0 1270 170' }>
+											<AsyncLoad
+												require="calypso/components/jetpack-header/woocommerce"
+												darkColorScheme={ false }
+												placeholder={ null }
+											/>
+										</svg>
+									</div>
 								</div>
-							</div>
-						) }
-					</Fragment>
-				);
-				headerText = translate( 'Log in with a WordPress.com account' );
-				postHeader = (
-					<p className="login__header-subtitle">
-						{ translate(
-							'Log in to WooCommerce.com with your WordPress.com account to connect your store and manage your extensions'
-						) }
-					</p>
-				);
+							) }
+						</Fragment>
+					);
+					headerText = translate( 'Log in with a WordPress.com account' );
+					postHeader = (
+						<p className="login__header-subtitle">
+							{ translate(
+								'Log in to WooCommerce.com with your WordPress.com account to connect your store and manage your extensions'
+							) }
+						</p>
+					);
+				} else {
+					preHeader = <Gridicon icon="my-sites" size={ 72 } />;
+					postHeader = (
+						<p>
+							{ translate(
+								'WooCommerce.com now uses WordPress.com Accounts.{{br/}}{{a}}Learn more about the benefits{{/a}}',
+								{
+									components: {
+										a: (
+											<a
+												href="https://woocommerce.com/2017/01/woocommerce-requires-wordpress-account/"
+												target="_blank"
+												rel="noopener noreferrer"
+											/>
+										),
+										br: <br />,
+									},
+								}
+							) }
+						</p>
+					);
+				}
```

Comment:
Did a bit of a reorg while adding the `isPartnerSignup` section.

From 

```
if ( isWooOAuth2Client( oauth2Client ) && ! wccomFrom ) {
	//
}
if ( isWooOAuth2Client( oauth2Client ) && wccomFrom ) {
	//
}
```

To

```
if ( isWooOAuth2Client( oauth2Client ) ) {
	if ( isPartnerSignup ) {
		//
	} else if ( wccomFrom ) {
		//
	} else {
		//
	}
}
```

Reply:
thanks for the clarification in here!